# Analysis of Visa Data

## Loading the First Dataset

In [2]:

import pandas as pd

# Load the first dataset
file_path = 'visas_dataF.csv'
visas_data = pd.read_csv(file_path)

# Display basic information about the dataset 
visas_data.info(), visas_data.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 18 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   Year                        10 non-null     object
 1   India                       10 non-null     int64 
 2   United Kingdom              10 non-null     int64 
 3   United States of America    10 non-null     int64 
 4   Singapore                   10 non-null     int64 
 5   People's Republic of China  10 non-null     int64 
 6   Malaysia                    10 non-null     int64 
 7   France                      10 non-null     int64 
 8   Germany                     10 non-null     int64 
 9   Indonesia                   10 non-null     int64 
 10  Philippines                 10 non-null     int64 
 11  Canada                      10 non-null     int64 
 12  Vietnam                     10 non-null     int64 
 13  Republic of Korea           10 non-null     int64 
 1

(None,
       Year   India  United Kingdom  United States of America  Singapore  \
 0  2012–13  133566          545378                    381197     170133   
 1  2013–14  151405          548485                    390102     194444   
 2  2014–15  175343          557655                    426856     203295   
 3  2015–16  189221          586174                    466030     236516   
 4  2016–17  220952          622869                    526605     244388   
 
    People's Republic of China  Malaysia  France  Germany  Indonesia  \
 0                      475589    215868  112602   140059      89209   
 1                      551937    267379  119986   145810      90134   
 2                      661855    292263  129967   151150      86086   
 3                      801761    327383  135984   159299      89403   
 4                      888584    356010  141759   173871     111187   
 
    Philippines  Canada  Vietnam  Republic of Korea  Nepal  Ireland   Other1  \
 0        44441  1027

## Calculating Probabilities for the First Dataset

In [3]:

# Creating a new DataFrame to store the probabilities
visas_probabilities = visas_data.copy()

# Calculate the probabilities correctly
for column in visas_data.columns:
    if column not in ['Year', 'Total']:
        visas_probabilities[column] = visas_data[column] / visas_data['Total']

# Display the first few rows of the corrected DataFrame containing the probabilities
visas_probabilities.head()

Year     India  United Kingdom  United States of America  Singapore  \
0  2012–13  0.035612        0.145409                  0.101635   0.045361   
1  2013–14  0.037944        0.137456                  0.097764   0.048730   
2  2014–15  0.040701        0.129443                  0.099082   0.047189   
3  2015–16  0.039428        0.122141                  0.097106   0.049283   
4  2016–17  0.041333        0.116518                  0.098510   0.045717   

   People's Republic of China  Malaysia    France   Germany  Indonesia  \
0                    0.126802  0.057555  0.030022  0.037343   0.023785   
1                    0.138321  0.067008  0.030070  0.036542   0.022589   
2                    0.153630  0.067840  0.030168  0.035085   0.019982   
3                    0.167062  0.068217  0.028335  0.033193   0.018629   
4                    0.166225  0.066598  0.026518  0.032525   0.020799   

   Philippines    Canada   Vietnam  Republic of Korea     Nepal   Ireland  \
0     0.011849  0.027387  0.008047           0.043376  0.001473  0.013433   
1     0.011661  0.025990  0.008123           0.041531  0.001841  0.012404   
2     0.012260  0.025404  0.009280           0.042960  0.002001  0.011708   
3     0.014091  0.023453  0.010279           0.045486  0.002273  0.010770   
4     0.016169  0.025040  0.011842           0.047536  0.002680  0.009919   

     Other1    Total  
0  0.290912  3750640  
1  0.282026  3990248  
2  0.273266  4308107  
3  0.270256  4799173  
4  0.272069  5345684

## Loading the Second Dataset

In [5]:

# Load the second dataset
file_path_new = 'visas_data1F.csv'
visas_data_new = pd.read_csv(file_path_new)

# Display basic information about the new dataset and the first few rows
visas_data_new.info(), visas_data_new.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 25 columns):
 #   Column                                     Non-Null Count  Dtype  
---  ------                                     --------------  -----  
 0   Year                                       10 non-null     object 
 1   Temporary Resident (Skilled Employment)    10 non-null     float64
 2   Visitors                                   10 non-null     float64
 3   Temporary Graduate                         10 non-null     float64
 4   Student                                    10 non-null     float64
 5   Working Holiday Maker                      10 non-null     float64
 6   All other visas                            10 non-null     float64
 7   No previous visa held or known             10 non-null     float64
 8   Total                                      10 non-null     int64  
 9   Temporary Resident (Skilled Employment).1  10 non-null     float64
 10  Visitors.1                   

(None,
       Year  Temporary Resident (Skilled Employment)   Visitors  \
 0  2012–13                                 7.077424  33.974487   
 1  2013–14                                 7.927236  33.128161   
 2  2014–15                                 8.566614  30.835216   
 3  2015–16                                 8.302703  30.547027   
 4  2016–17                                 7.666530  31.487536   
 
    Temporary Graduate   Student  Working Holiday Maker  All other visas  \
 0            0.319626  3.978197               2.942268         1.760794   
 1            0.948759  5.441643               2.595366         3.344180   
 2            0.892276  4.300540               2.313024         4.553288   
 3            0.419459  3.662703               2.417297         5.171892   
 4            0.269718  2.897425               2.223130         5.888843   
 
    No previous visa held or known  Total  \
 0                       49.947205    100   
 1                       46.614655    100

## Calculating Probabilities for 'Working Holiday Maker' Visa

In [6]:

# Extract the Visitors.2 and Working Holiday Maker.2 columns to get P(V) and P(W) respectively.
P_V = visas_data_new['Visitors.2'] / 100  # Dividing by 100 to convert percentage to probability
P_W = visas_data_new['Working Holiday Maker.2'] / 100  # Dividing by 100 to convert percentage to probability

# Create a new DataFrame to store the probabilities of a person from a certain country applying for a Working Holiday Maker visa.
working_holiday_maker_probabilities = visas_data[['Year']].copy()

# Calculate P(W | C) for each country in each year.
for column in visas_data.columns:
    if column not in ['Year', 'Total']:
        P_C = visas_probabilities[column]  # Extract P(C) from the visas_probabilities DataFrame
        P_W_C = P_C * (P_W / P_V)  # Calculate P(W | C) using the provided formula.
        working_holiday_maker_probabilities[column] = P_W_C  # Store the calculated probabilities in the new DataFrame.

# Display the first few rows of the new DataFrame containing the calculated probabilities for Working Holiday Maker visa.
working_holiday_maker_probabilities.head()

Year     India  United Kingdom  United States of America  Singapore  \
0  2012–13  0.023859        0.097419                  0.068092   0.030390   
1  2013–14  0.025525        0.092466                  0.065765   0.032780   
2  2014–15  0.025400        0.080780                  0.061833   0.029449   
3  2015–16  0.023359        0.072361                  0.057530   0.029197   
4  2016–17  0.023022        0.064899                  0.054869   0.025464   

   People's Republic of China  Malaysia    France   Germany  Indonesia  \
0                    0.084953  0.038560  0.020114  0.025018   0.015935   
1                    0.093048  0.045076  0.020228  0.024581   0.015195   
2                    0.095874  0.042336  0.018827  0.021895   0.012470   
3                    0.098974  0.040414  0.016787  0.019665   0.011036   
4                    0.092585  0.037094  0.014770  0.018116   0.011585   

   Philippines    Canada   Vietnam  Republic of Korea     Nepal   Ireland  \
0     0.007938  0.018348  0.005391           0.029060  0.000987  0.008999   
1     0.007844  0.017483  0.005464           0.027938  0.001239  0.008344   
2     0.007651  0.015853  0.005791           0.026810  0.001249  0.007306   
3     0.008348  0.013894  0.006090           0.026948  0.001347  0.006380   
4     0.009006  0.013947  0.006596           0.026477  0.001493  0.005525   

     Other1  
0  0.194901  
1  0.189717  
2  0.170534  
3  0.160110  
4  0.151539



- \( P(C) \) as the probability of a person from a certain country (from the first dataset).
- \( P(V) \) as the percentage of people who applied for a "Visitor" visa (from the `Visitors.2` column in the second dataset).
- \( P(W) \) as the percentage of people who applied for a "Working Holiday Maker" visa (from the `Working Holiday Maker.2` column in the second dataset).
- \( P(W | C) \) as the probability of a person from a certain country applying for a "Working Holiday Maker" visa.

Given that we know the probability of a person from a certain country applying for a "Visitor" visa, we can use the proportions of "Visitor" and "Working Holiday Maker" visas to find the probability of a person from a certain country applying for a "Working Holiday Maker" visa.

The relationship between these probabilities can be expressed as follows:

$  P(W | C) = P(C) \times \left( \frac{P(W)}{P(V)} \right) $

If we calculate \( P(W | C) \) for each country and each year.

Now we have the probabilities of a person from a certain country applying for a "Working Holiday Maker" visa for each year. Here are the first few rows of the calculated probabilities:


\begin{array}{|l|l|l|l|l|l|l|l|l|l|l|l|l|l|l|l|l|}
\hline
\text{Year} & \text{India} & \text{United Kingdom} & \text{United States of America} & \text{Singapore} & \text{People's Republic of China} & \text{Malaysia} & \text{France} & \text{Germany} & \text{Indonesia} & \text{Philippines} & \text{Canada} & \text{Vietnam} & \text{Republic of Korea} & \text{Nepal} & \text{Ireland} & \text{Other1} \\
\hline
2012–13 & 0.0239 & 0.0974 & 0.0681 & 0.0304 & 0.0850 & 0.0386 & 0.0201 & 0.0250 & 0.0159 & 0.0079 & 0.0183 & 0.0054 & 0.0291 & 0.0010 & 0.0090 & 0.1949 \\
2013–14 & 0.0255 & 0.0925 & 0.0658 & 0.0328 & 0.0930 & 0.0451 & 0.0202 & 0.0246 & 0.0152 & 0.0078 & 0.0175 & 0.0055 & 0.0279 & 0.0012 & 0.0083 & 0.1897 \\
2014–15 & 0.0254 & 0.0808 & 0.0618 & 0.0294 & 0.0959 & 0.0423 & 0.0188 & 0.0219 & 0.0125 & 0.0077 & 0.0159 & 0.0058 & 0.0268 & 0.0012 & 0.0073 & 0.1705 \\
2015–16 & 0.0234 & 0.0724 & 0.0575 & 0.0292 & 0.0990 & 0.0404 & 0.0168 & 0.0197 & 0.0110 & 0.0083 & 0.0139 & 0.0061 & 0.0269 & 0.0013 & 0.0064 & 0.1601 \\
2016–17 & 0.0230 & 0.0649 & 0.0549 & 0.0255 & 0.0926 & 0.0371 & 0.0148 & 0.0181 & 0.0116 & 0.0090 & 0.0139 & 0.0066 & 0.0265 & 0.0015 & 0.0055 & 0.1515 \\
\hline
\end{array}

